nixos/modules/installer/tools/nixos-rebuild.sh

```diff
@@ -164,16 +255,16 @@ if [ -n "$buildNix" ]; then
                     echo "$0: unsupported platform"
                     exit 1
                 fi
-                if ! nix-store -r $nixStorePath --add-root $tmpDir/nix --indirect \
+                if ! buildHostCmd nix-store -r $nixStorePath --add-root $tmpDir/nix --indirect \
                     --option extra-binary-caches https://cache.nixos.org/; then
                     echo "warning: don't know how to get latest Nix" >&2
                 fi
                 # Older version of nix-store -r don't support --add-root.
-                [ -e $tmpDir/nix ] || ln -sf $nixStorePath $tmpDir/nix
+                buildHostCmd [ -e $tmpDir/nix ] || ln -sf $nixStorePath $tmpDir/nix
             fi
         fi
     fi
-    PATH=$tmpDir/nix/bin:$PATH
+    PATH="$tmpDir/nix/bin:$PATH"
```

Comment:
The whole point of evaluating the Nix interpreter before the configuration, is to ensure that we have the lastest version for instantiating and building.  If we specify a build host, then we probably want to use an update-to-date version of Nix as well for evaluating the configuration.  In this particular case, having a build host set would set the path which does not exists locally.